packages/esm-patient-flags-app/src/hooks/usePatientFlags.tsx

```diff
@@ -0,0 +1,51 @@
+import { openmrsFetch } from '@openmrs/esm-framework';
+import { useEffect, useMemo, useState } from 'react';
+import useSWR from 'swr';
+
+interface FlagFetchResponse {
+  uuid: string;
+  name: string;
+  message: string;
+}
+
+interface FlagsFetchResponse {
+  results: Array<FlagFetchResponse>;
+}
+
+/**
+ * React hook that takes in a patient uuid and returns
+ * patient flags for that patient together with helper objects
+ * @param patientUuid Unique patient idenfier
+ * @returns An array of patient identifiers
+ */
+export function useFlagsFromPatient(patientUuid: string) {
+  const { data, error, isValidating, mutate } = useSWR<{ data: FlagsFetchResponse }, Error>(
+    `/ws/rest/v1/patientflags/flag?patient=${patientUuid}&v=full`,
+    openmrsFetch,
+  );
+  const result = useMemo(() => {
+    return {
+      flags: data?.data?.results ?? null,
+      flagLoadingError: error,
+      isLoadingFlags: !data && !error,
+      isValidatingFlags: isValidating,
+      mutate,
+    };
+  }, [data, error, isValidating, mutate]);
+  return result;
+}
+
+export function useCurrentPath(): string {
+  const [path, setPath] = useState(window.location.pathname);
+  const listenToRoutingEvent = () => {
+    const winPath = window.location.pathname;
+    setPath(winPath);
+  };
+  useEffect(() => {
```

Comment:
Please make either of the choice below:
1. Move `listenToRoutingEvent` inside the useEffect
2. Use `useCallback` for the `listenToRoutingEvent` and add the function as the dependency for the useEffect.

Reply:
Reason being, on every render the function's reference will change and the useEffect will take the initial reference only, hence we either use `useCallback` or shift it inside the useEffect.